drivers: eth: stellaris: Fix driver poll all packtes from ethernet

su47flying opened the pull request:
The number of interrupts received by the driver maybe less than the number of data packets received by Ethernet,
so the driver should read the packets using the 'np' value.

jukkar commented on drivers/ethernet/eth_stellaris.c:
```diff
@@ -219,19 +219,21 @@ static void eth_stellaris_rx(const struct device *dev)
 {
 	struct eth_stellaris_runtime *dev_data = dev->data;
 	int isr_val = sys_read32(REG_MACRIS);
+	int np = 0;
```
Setting value for this variable here is not needed as we are setting it already below.

decsny commented on drivers/ethernet/eth_stellaris.c:
```diff
@@ -240,7 +242,18 @@ static void eth_stellaris_isr(const struct device *dev)
+		 * to get how many packets are in the Ethernet.
+		 */
+		np = sys_read32(REG_MACNP);
+		for(int i = 0; i < np; i++) {
```
please fix the compliance failure found by the ci runner

decsny commented on drivers/ethernet/eth_stellaris.c:
```diff
@@ -219,19 +219,21 @@ static void eth_stellaris_rx(const struct device *dev)
 {
 	struct eth_stellaris_runtime *dev_data = dev->data;
 	int isr_val = sys_read32(REG_MACRIS);
+	int np = 0;
```
instead of calling the variable `np`, call it something more verbose like `num_packets`, the meaning is more obvious then

su47flying replied:
When the device or qemu receives more packets, it sends all packets to the driver and try send more interrupts to driver.
However, if the driver does not ACK the interrupt, the interrupt is in a suspended state and the device cannot send the interrupt to the driver.

decsny replied:
okay, well the lock is not new anyways, so if you understand what will happen then I will assume it doesn't cause any issue

su47flying replied:
Interrupts are a notification, not an assumption that it's equal to the number of packets.
Therefore, device drivers (e.g., e1000, virtio-net) all need to read the data length and retrieve data from the device based on that length.
So, it doesn't cause any issue.

decsny reviewed (changes requested):
out of curiosity, do you know what would happen if the module gets more packets during the period while the interrupts are locked and processing the packets from the last interrupt? would the packets received during the interrupt lock period be granted a second interrupt after the unlock even if they were finished being received before then? let me know if my question is not clear

jukkar reviewed (approved):
LGTM

Commit:
drivers: eth: stellaris: Fix driver poll all packets from ethernet

The number of interrupts received by the driver maybe less than
the number of data packets received by Ethernet,
so the driver should read the packets number
from the register REG_MACNP value.

Signed-off-by: Wafer Xie <wafer@jaguarmicro.com>
